File: JavaDecoder/app/src/main/java/com/wolt/blurhash/MainActivity.java
package com.wolt.blurhash;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Bitmap;
import android.os.Bundle;
import android.os.SystemClock;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;

import com.wolt.blurhash.decoder.BlurHashDecoder;

/**
 * @author Ilanthirayan Paramanathan <theebankala@gmail.com>
 * @version 2.0.0
 * @since 22nd of November 2020
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        EditText etInput = findViewById(R.id.etInput);
        TextView tvDecode = findViewById(R.id.tvDecode);
        ImageView ivResult = findViewById(R.id.ivResult);
        TextView ivResultTime = findViewById(R.id.ivResultTime);

        tvDecode.setOnClickListener(view -> {
            long tStart = SystemClock.elapsedRealtime();
            Bitmap bitmap = BlurHashDecoder.getInstance().decode(etInput.getText().toString(), 20, 12, 1.0F, true);
            long tEnd = SystemClock.elapsedRealtime();
            long tRes = tEnd - tStart; // time in milliseconds
            ivResult.setImageBitmap(bitmap);
            ivResultTime.setText(String.format("Time : %dms", tRes));
        });
    }
}

File: Kotlin/demo/src/main/java/com/wolt/blurhashapp/MainActivity.kt
package com.wolt.blurhashapp

import android.graphics.Bitmap
import android.os.Bundle
import android.os.SystemClock
import androidx.appcompat.app.AppCompatActivity
import com.wolt.blurhashapp.databinding.ActivityMainBinding
import com.wolt.blurhashkt.BlurHashDecoder


class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.tvDecode.setOnClickListener {
            var bitmap: Bitmap? = null
            val time = timed {
                bitmap = BlurHashDecoder.decode(binding.etInput.text.toString(), 20, 12)
            }
            binding.ivResult.setImageBitmap(bitmap)
            binding.ivResultTime.text = "Time: $time ms"
        }
    }

}

/**
 * Executes a function and return the time spent in milliseconds.
 */
private inline fun timed(function: () -> Unit): Long {
    val start = SystemClock.elapsedRealtime()
    function()
    return SystemClock.elapsedRealtime() - start
}

